cow_instrument/SpectrumInfo.h

```diff
@@ -0,0 +1,89 @@
+#ifndef SPECTRUMINFO_H
+#define SPECTRUMINFO_H
+
+#include <vector>
+#include "Spectrum.h"
+#include "DetectorInfo.h"
+#include "InstrumentTree.h"
+#include "L2s.h"
+
+namespace {
+
+template <typename U>
+void spectraRangeCheck(size_t spectraIndex, const U &container) {
+  if (spectraIndex >= container.size()) {
+    std::stringstream buffer;
+    buffer << "Spectra at index " << spectraIndex << " is out of range";
+    throw std::out_of_range(buffer.str());
+  }
+}
+}
+
+template <typename InstTree> class SpectrumInfo {
```

Comment:
I forgot why we need this as a template, but that is also related to my question above with the double-template.